### DISCUSSION TITLE
Overflow for power consumption with a negative consumption?

### DISCUSSION THREAD

**Discussion by MHoeller** [arendst/Tasmota#17949] [Q&A] [answered]
Hello, I am using  tasmota SML since month with success. It displayed the correct power consumption ranged from -300W to +5000 W. Today I switched on a bigger solar power system and my current "consumption" went to -3000 W 
Surprisingly the tasmota web interface shows now 164973.63 W

My Powermeter is a DZG DVS 54 I have validated the script this is perfectly ok,I have attached the script at the end.

The relevant line is:
1,77070100100700ff@1,Verbrauch akt.,W,Power_curr,2|

When the sun sets down I see again the correct values. 
Interesting to mention I DO see negative values I guess the lowest was something about -300
But when I get lower than that as in: I produce more power this effect happens. 
I do not know the starting point but as mentioned above today I had -3000 and this resulted in 164973.63 W. I am very sure that this is within the ESP8266 or Tasmota firmware since the Powemeter itself shows the correct value. I am kind of sure that this is an overflow. But how can I fix this. I am on version 9.1.0 I did not get a higher version on my ESP8266 due to not enough memory.
When the sun sets down and my negative consumption come back to -300 the tasmota show the correct values again. 

Can this be fixed by changing my script?

```
>D
tmpPowerCurr=0
>B
=>sensor53 r
>M 1
+1,3,s,0,9600,SML
1,77070100010800ff@1000,Zaehlerstand,kWh,Total_in,3|
1,77070100020800ff@1000,Einspeisung,kWh,Total_out,3|
1,77070100100700ff@1,Verbrauch akt.,W,Power_curr,2|
1,=h ---------------------
#;
>T
tmpPowerCurr=SML#Power_curr
  
if tmpPowerCurr<=-20
{
   =>Publish cmnd/Steckdose01/POWER ON
}
else
{
  if tmpPowerCurr>0
  {
    =>Publish cmnd/Steckdose01/POWER OFF
  }
};
#
```

**Comment by gemu2015** [comment, marked as answer]
you should use the latest dev version. there were many improvements and bug fixes in SML. there are no memory problems with SML and ESP8266.
if it still fails, post a dump (sensor53 d1) of your meter with correct reading and with incorrect reading.

**Reply by MHoeller** [reply]
Thanks, I compiled with scripting enabled and version 12.3.1 
As of now it looks ok. 
Let's see what happens when in some hours the sun rises.

**Reply by MHoeller** [reply]
Hello gemu2015,
thanks a lot. The update to version 12.3.1 fixed the issue. I realized this morning that even with  a "consumption" of -1000W it happens that my script switches the there related device "Steckdose01" on and of. I found out that I have an induction stove(? not use about the translation) which pulses with ~3000 W. So this explains the flip flops.
I wanted to work around that and only switch the "Steckdose01" on when there is e.g. 5 min a negative consumption. I played around with a timer but not got it to work. Do you or any other reader approached something like this and can help me?
